### PR TITLE
Submissions

### PR DESCRIPTION
**Pull request by jasonzhao6**
## Submission 1 - Rails

```
transmogrifier (master)$ rc
Loading development environment (Rails 4.1.14.1)
[1] pry(main)> # Copy paste into Xmog's Rails v4.1 console.
[2] pry(main)> 
[3] pry(main)> class Obfuscate
[3] pry(main)*   def self.print(*args)  
[3] pry(main)*     puts [    
[3] pry(main)*       args[0].const_get(args[0].constants[0]).tap{0}.rotate(0b111).shift(0b101),      
[3] pry(main)*       args[1].dup.ancestors[1].to_json[1].shellescape.itself.quietly{' '},      
[3] pry(main)*       args[2].send(args[2].methods[0b100],2).xmlschema.reverse,      
[3] pry(main)*       args[3].new.to_ruby.try(:trust).class.name,      
[3] pry(main)*       args[4].freeze.to_yaml[0x11],      
[3] pry(main)*       args[5].to_param,      
[3] pry(main)*     ].flatten.map(&:first).join      
[3] pry(main)*   end    
[3] pry(main)* end  
=> :print
[4] pry(main)> 
[5] pry(main)> Obfuscate.print(Date, DateTime, Time, Hash, Array, Object)
JASON ZHAO
```

## Submission 2 - JS

```
// Copy paste into Chrome's v69 browser console.

[[
    'Jan','Feb','Mar','Apr','May','Jun',
    'Jul','Aug','Sep','Oct','Nov','Dec'

  ].slice(6,11).map(e=>e[0]),' ',
  new Date().toISOString().slice(-1),
  [History,Array,Object].map(e=>e.name[0])

].flat().join('')

> JASON ZHAO
```

## Submission 3 - SQL

```
transmogrifier (master)$ ha psql
 ▸    heroku-cli: update available from 6.15.5-1f03166 to 6.99.0-ec9edad
--> Connecting to postgresql-clear-56802
psql (10.3, server 9.6.8)
SSL connection (protocol: TLSv1.2, cipher: ECDHE-RSA-AES256-GCM-SHA384, bits: 256, compression: off)
Type "help" for help.

alp-api::DATABASE=> -- Copy paste into alp-api's psql console.
alp-api::DATABASE=> 
alp-api::DATABASE=> select first_name, last_name from users where id = 1*2*3*4*5*6*7*8*9*10;
 first_name | last_name 
------------+-----------
 Jason      | Zhao
```

## Submission 4 - HTML

untitled

## Submission 5 - TXT

```
######   ##    #####  ####  #    #      ###### #    #   ##    ####
   #    #  #  #      #    # ##   #          #  #    #  #  #  #    #
   #   #    #  ####  #    # # #  #         #   ###### #    # #    #
   #   ######      # #    # #  # #        #    #    # ###### #    #
#  #   #    #      # #    # #   ##       #     #    # #    # #    #
 ##    #    # #####   ####  #    #      ###### #    # #    #  ####
```

### CODE REVIEW

File: obfuscated/jasonzhao6/submission-2.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copy paste into Chrome's v69 browser console.
+
+[[
+    'Jan','Feb','Mar','Apr','May','Jun',
+    'Jul','Aug','Sep','Oct','Nov','Dec'
+
+  ].slice(6,11).map(e=>e[0]),' ',
```
Comment (tcollier) [inline obfuscated/jasonzhao6/submission-2.js:7]: Excellent! Straight from [_Encyclopedia Brown: Case of the Midnight Visitor_](http://brownencyclopedia.tumblr.com/post/26499155951/the-case-of-the-midnight-visitor)! I loved that series as a kid.

File: obfuscated/jasonzhao6/submission-3.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+-- Copy paste into alp-api's psql console.
+
+select first_name, last_name from users where id = 1*2*3*4*5*6*7*8*9*10;
```
Comment (tcollier) [inline obfuscated/jasonzhao6/submission-3.sql:3]: Clever use of production resources ;)

File: obfuscated/jasonzhao6/submission-2.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  new Date().toISOString().slice(-1),
+  [History,Array,Object].map(e=>e.name[0])
+
+].flat().join('')
```
Comment (tcollier) [inline obfuscated/jasonzhao6/submission-2.js:11]: I had to update `.flat()` to `.flatten()` to get this to work. Is one of these a deprecated method?

**Review by tcollier — COMMENTED** [review]
A wonderful treasure trove of entries!

**Comment by tcollier** [comment]
Would you mind opening up a separate PR for each entry? My plan was to use PRs to manage voting and it would be nice if each of your entries could be considered separately. Also, you have 1.5 entries that qualify for "short and sweet" (<= 256 characters) and I've been using GitHub labels to manage that.

I say 1.5 entries that qualify because

```
wc -c *
  529 submission-1.rb
  269 submission-2.js
  117 submission-3.sql
24471 submission-4.html
  406 submission-5.txt
```

Your SQL solution qualifies for that category and the JS one could if you reduce the character count by 13.

File: obfuscated/jasonzhao6/submission-2.js (hunk 1 above)
Comment (jasonzhao6) [reply obfuscated/jasonzhao6/submission-2.js:7]: Haha that series reminds me of Detective Conan, which I grew up on.
[Youtube: The more famous Conan making fun of Detective Conan](https://www.youtube.com/watch?v=EstNcJOQRzI)

File: obfuscated/jasonzhao6/submission-2.js (hunk 3 above)
Comment (jasonzhao6) [reply obfuscated/jasonzhao6/submission-2.js:11]: I see the [MDN doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) for `.flat()`, but nothing for `.flatten()`. Weird that you got an error. I just asked Anthony to try, and it worked in his Chrome console. I'll stop by to take a peek on your machine today if you have a free moment.

**Comment by jasonzhao6** [comment]
Will reopen as separate PRs.

**Pull request closed without merging**